ttnn/cpp/ttnn/tensor/xtensor/partition.hpp

```diff
@@ -0,0 +1,27 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "ttnn/tensor/tensor.hpp"
+#include "ttnn/tensor/xtensor/xtensor_all_includes.hpp"
+
+namespace ttnn::experimental::xtensor {
+
+// IMPORTANT: `chunk` and `concatenate` are not yet part of the public ttnn API. Internally, they rely on the `xtensor`
+// library for efficient host-side operations.
+
+// Splits a tensor into chunks along the specified dimension.
+std::vector<tt::tt_metal::Tensor> chunk(const tt::tt_metal::Tensor& tensor, int num_chunks, int dim = 0);
+
+template <typename T>
+std::vector<xt::xarray<T>> chunk(const xt::xarray<T>& tensor, int num_chunks, int dim = 0);
+
+// Concatenates a list of tensors along the specified dimension.
+tt::tt_metal::Tensor concatenate(const std::vector<tt::tt_metal::Tensor>& tensors, int dim = 0);
```

Comment:
consider `concat` or `join`

Reply:
Two other apis operate on xt::xarray, but this one operates on a Tensor. This seem a bit off to me

Reply:
maybe (just maybe) this is another Tensor constructor

Reply:
> Two other apis operate on xt::xarray, but this one operates on a Tensor.

Yeah, likely we will move tt-train off the one that relies on `xt::array`. I'm not too worried for now, as both of these rely on the same impl.

I'm not sure about the ctor, this feels more like an utility function (which also shout sit next to `chunk` IMO). Big +1 that we need to have a better organization though.

And sure, `concat` sounds good. No strong opinion here.